### PR TITLE
Add architecture diagram and example module usage

### PR DESCRIPTION
**Pull request by lxuo-trussworks**
This PR adds:

1. an architecture diagram of the deployment to AWS and it's interaction with Github
2. an example of the module usage in a terraform deployment

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -84,6 +86,33 @@ All of the specified resources in the IAM policy do not have to exist prior to t
...
+Here is an example of how you would use the module in your own terraform deployment:
+
+```text
+module "github-actions-runner-aws" {
+  source = "github.com/cmsgov/github-actions-runner-aws?ref=v1.0.0"
+
+  # ECR variables
+  container_name          = "github-runner"
+  allowed_read_principals = local.principals_ro_access
+  ci_user_arn             = data.aws_iam_user.github_ci_user.arn
```
Comment (dynamike) [inline README.md:100]: Can you include the example of how this is defined?

File: README.md [hunk 2]
```diff
@@ -84,6 +86,33 @@ All of the specified resources in the IAM policy do not have to exist prior to t
...
+
+  # ECS variables
+  environment               = "dev"
+  ecs_desired_count         = 0
```
Comment (dynamike) [inline README.md:104]: Maybe it makes sense to remove `ecs_desired_count` as a variable now that the count is being managed by the actions workflow

File: README.md [hunk 3]
```diff
@@ -84,6 +86,33 @@ All of the specified resources in the IAM policy do not have to exist prior to t
...
+  ecs_vpc_id                = data.aws_vpc.example_east_sandbox.id
```
Comment (dynamike) [inline README.md:105]: Can you include example of how the VPC is being looked up as a data source?

File: README.md [hunk 4]
```diff
@@ -84,6 +86,33 @@ All of the specified resources in the IAM policy do not have to exist prior to t
...
+  ecs_subnet_ids            = data.aws_subnet_ids.private.ids
+  ecr_repo_tag              = "latest"
+  logs_cloudwatch_group_arn = aws_cloudwatch_log_group.main.arn
+
+  # GitHub Runner variables
+  personal_access_token_arn = data.aws_secretsmanager_secret_version.token.arn
```
Comment (dynamike) [inline README.md:111]: Can you include example of how the secret is being looked up as a data source?

File: README.md [hunk 5]
```diff
@@ -84,6 +86,33 @@ All of the specified resources in the IAM policy do not have to exist prior to t
 }
 ```
 
+## Terraform Deployment
+
+Here is an example of how you would use the module in your own terraform deployment:
+
+```text
+module "github-actions-runner-aws" {
+  source = "github.com/cmsgov/github-actions-runner-aws?ref=v1.0.0"
```
[the remark this answers is not in the document]
Comment (lxuo-trussworks) [reply README.md:116]: Looked into this and discussed with Adam a bit. We could do it, but we don't really need to do it right now -- so going to skip for now, but possibly revisit in the future. I did go ahead with updating the docs a bit to incorporate documentation we'd include with pushing it to Terraform's registry, though.

[the remark this answers is not in the document]
Comment (dynamike) [reply README.md:116]: That's fine with me. There's a bit more nuance to this repo given that it also builds and publishes the docker image too.

File: README.md (hunk 1 above)
Comment (lxuo-trussworks) [reply README.md]: just added to the docs for all of these, thanks for pointing this out

File: README.md (hunk 2 above)
Comment (lxuo-trussworks) [reply README.md:125]: I think it's still worth having the flexibility. it is optional, and with the updated docs, it's more clear it's optional/has a default of 0

File: README.md [hunk 6]
```diff
@@ -84,6 +86,33 @@ All of the specified resources in the IAM policy do not have to exist prior to t
...
+
+  # ECR variables
+  container_name          = "github-runner"
+  allowed_read_principals = local.principals_ro_access
```
[the remark this answers is not in the document]
Comment (lxuo-trussworks) [reply README.md]: updated for this to fall more in-line with being a registered module

File: README.md (hunk 3 above)
Comment (lxuo-trussworks) [reply README.md]: I ended up making this usage more generic, but did add a section on how to use data sources to look-up info in AWS

File: README.md (hunk 4 above)
Comment (lxuo-trussworks) [reply README.md]: yup, added under data source example section

**Comment by chtakahashi** [comment]
:shipit:

**Review by chtakahashi — APPROVED** [review]

**Pull request merged**